module/Photo/src/Photo/Model/Photo.php

```diff
@@ -334,14 +333,26 @@ public function setPath($path)
     }
 
     /**
-     * Updates the photoCount in the album object.
+     * Updates the photoCount and date in the album object.
      * 
      * @ORM\PrePersist()
      * @ORM\PostUpdate() 
      */
-    public function incrementOnAdd()
+    public function updateOnAdd()
     {
         $this->album->setPhotoCount($this->album->getPhotoCount() + 1);
+        //update start and end date if the added photo is newere or older
+        if (is_null($this->album->getStartDateTime())) {
+            $this->album->setStartDateTime($this->getDateTime());
+        } else if ($this->album->getStartDateTime()->getTimestamp() > $this->getDateTime()->getTimeStamp()) {
+            $this->album->setStartDateTime($this->getDateTime());
+        }
+
+        if (is_null($this->album->getEndDateTime())) {
+            $this->album->setEndDateTime($this->getDateTime());
+        } else if ($this->album->getStartDateTime()->getTimestamp() < $this->getDateTime()->getTimeStamp()) {
+            $this->album->setEndDateTime($this->getDateTime());
+        }
```

Comment:
Dat is inderdaad een beter idee. Eerst hadden we verschillende code in beide conditions, maar dat is nu niet meer het geval.